integration_test/third_party_apps_data/applications/dcgm/centos_rhel/install

```diff
@@ -1,54 +1,81 @@
-# less errors on installation; fallback to the runfile method if the package 
-# manager's package is not working or not compitible with the GPU model
-DEVICE_CODE=$(lspci -n | grep -Po '10de:[\w\d]{4}')
-case $DEVICE_CODE in
-    10de:102d)
-        # Install a specific version for NVIDIA Tesla K80, R470 is the last supported version
-        DRIVER_VERSION=470.82.01
-        ;;
-    *)
-        # Installing latest version of NVIDIA CUDA and driver
-        DRIVER_VERSION=535.129.03
-        ;;
-esac
+VERSION_ID=${VERSION_ID%%.*}
 
-echo "Installing NVIDIA Data Center driver $DRIVER_VERSION"
-curl -fSsl -O https://us.download.nvidia.com/tesla/$DRIVER_VERSION/NVIDIA-Linux-x86_64-$DRIVER_VERSION.run
-sudo bash ./NVIDIA-Linux-x86_64-$DRIVER_VERSION.run --silent
+install_driver_from_runfile() {
+    # Ref: https://docs.nvidia.com/datacenter/tesla/tesla-installation-notes/index.html#runfile
+    # This method requires the match kernel-devel package to be installed, and 
+    # the package may be absent from the repo and cause this method to fail
+    sudo yum install -y kernel-devel-$(uname -r) pciutils gcc make wget yum-utils 
+    local DRIVER_VERSION=535.129.03
+    echo "Installing NVIDIA Data Center driver $DRIVER_VERSION"
+    curl -fSsl -O https://us.download.nvidia.com/tesla/$DRIVER_VERSION/NVIDIA-Linux-x86_64-$DRIVER_VERSION.run
+    sudo bash ./NVIDIA-Linux-x86_64-$DRIVER_VERSION.run --silent
 
-# check NVIDIA driver installation succeeded
-nvidia-smi
+    # check NVIDIA driver installation succeeded
+    nvidia-smi
+}
 
-# Install DCGM
-VERSION_ID=${VERSION_ID%%.*}
-sudo yum-config-manager \
-    --add-repo http://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
-sudo yum clean expire-cache
-sudo yum install -y datacenter-gpu-manager
-sudo systemctl --now enable nvidia-dcgm
-
-# check DCGM service running and load profiling module
-dcgmi discovery --list
+install_dcgm() {
+    # Ref: https://docs.nvidia.com/datacenter/dcgm/latest/user-guide/getting-started.html#rhel-centos-rocky-linux
+    sudo yum install -y yum-utils
+    sudo yum-config-manager \
+    --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
+    sudo yum clean expire-cache
+    sudo yum install -y datacenter-gpu-manager
+    sudo systemctl --now enable nvidia-dcgm 
+
+    # check DCGM service running and load profiling module
+    dcgmi discovery --list
+}
+
+try_install() {
+    for install_method in "$@"
+    do
+        echo "Installing NVIDIA driver with $install_method..."
+        # Run the installation method in a subshell with -e so that the function
+        # exit on error
+        set +e ; ( set -e; $install_method ) ; exit_code=$?
+        set -e
+        if [ $exit_code -eq 0 ]; then
+            echo "NVIDIA driver has been installed successfully with $install_method."
+            return 
+        fi
+    done
+    return 1
```

Comment:
This return code is not checked anywhere. Do we need to return it?

Reply:
My thought here is to have this return code works together with `set -e` - so that if none of the installation methods works, the script stops after `try_install` and do not run following commands.

Reply:
I see. Makes sense. I would then change line 50 to `return 0` to match.

Reply:
Done!